### ISSUE TITLE
Feature request: prefer `find` over `filter(...)[0]`

### ISSUE THREAD

**Issue by kangax** [wix/eslint-plugin-lodash#175]
It'd be nice to suggest using `find(predicate)` over `filter(predicate)[0]` which I came across in our code.

**Comment by ganimomer** [comment]
Great idea!
Just to formalize the spec:
```js
//Bad:
_.filter(arr, predicate)[0]; // prefer using `_.find` over getting the first item of `filter`

_.head(_.filter(arr, predicate)); // prefer using `_.find` over getting the first item of `filter`

_(arr)
 .filter(predicate)
  .head() // prefer using `_.find` over getting the first item of `filter`

_.last(_.filter(arr, predicate)) // prefer using `_.findLast` over getting the last item of `filter`

_(arr)
  .filter(predicate)
  .last() // prefer using `_.findLast` over getting the last item of `filter`

// Good:
_.filter(arr, predicate)[i]

_.find(arr, predicate)
```
Alright by you?

**Comment by kangax** [comment]
Yes, this looks great. I’ll see if I can find some time for this or ask one of our engineers.

**Comment by ganimomer** [comment]
Added in `v2.7.0`.

**Comment by kangax** [comment]
Awesome, thank you!
